backend/src/common/util/workflow.go

```diff
@@ -202,15 +202,19 @@ func (w *Workflow) OverrideName(name string) {
 	w.Name = name
 }
 
-// SetAnnotations sets annotations on all templates in a Workflow
-func (w *Workflow) SetAnnotationsToAllTemplates(key string, value string) {
+// SetAnnotationsToAllTemplatesIfKeyNotExist sets annotations on all templates in a Workflow
+// if the annotation key does not exist
+func (w *Workflow) SetAnnotationsToAllTemplatesIfKeyNotExist(key string, value string) {
 	if len(w.Spec.Templates) == 0 {
 		return
 	}
 	for index := range w.Spec.Templates {
 		if w.Spec.Templates[index].Metadata.Annotations == nil {
 			w.Spec.Templates[index].Metadata.Annotations = make(map[string]string)
 		}
+		if _, isSet := w.Spec.Templates[index].Metadata.Annotations[key]; isSet {
+			return
```

Comment:
nit: I believe you want to write `continue` here, is this a mistake?

Reply:
Yes, you are right. This made our pipeline's steps always use Istio and broke some ops using Workload Identity yesterday. I fixed locally and forgot. Sorry for my broken patch.

I confirmed this patch has no diff from our latest env now.

Reply:
BTW, the failure of Workload Identity was fixed by annotating `"traffic.sidecar.istio.io/excludeOutboundIPRanges=169.254.169.254/32"`.

https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#fails

@Bobgy 🙇‍♂️